c7n/query.py

```diff
@@ -751,6 +751,8 @@ def __repr__(cls):
             identifier = cls.config_type
         elif cls.cfn_type:
             identifier = cls.cfn_type
+        elif cls.__doc__:
```

Comment:
what's this one for?

Reply:
without it, the logic was going to the default `identifier = "AWS::%s::%s" % (cls.service.title(), cls.id)` for `class TypeInfo`, but `class TypeInfo` has no `service`, this was the only workaround I found

what do you advise?

Reply:
gotcha, just trying to understand the logic

Reply:
fwiw, went ahead and updated this handling to not interrupt existing subclass identification.